### PR TITLE
Add AI onboarding ARM templates

### PR DESCRIPTION
**Pull request by jatakiajanvi12**
This pull request introduces two new JSON files for Azure Resource Manager (ARM) templates, which are used to automate deployments in Microsoft Azure. The first file, `existingClusterOnboarding.json`, defines the deployment template for an existing Azure Kubernetes Service (AKS) cluster. The second file, `existingClusterParam.json`, provides parameter values for the deployment template.

Here are the key changes:

* [`AIOnboardingARMTemplates/existingClusterOnboarding.json`](diffhunk://#diff-5ae6a92df76f06138ee7303ac1749a16992396c6be31d302dcce240faf06f63cR1-R89): This new file is an ARM template for onboarding an existing AKS cluster. It includes parameters for the AKS resource ID and location, ports for OpenTelemetry logs and metrics, and tags for AKS, ContainerInsights, and DataCollectionRule resources. The template also includes a deployment for an AKS monitoring add-on, which enables auto-instrumentation and OpenTelemetry logs and metrics.

* [`AIOnboardingARMTemplates/existingClusterParam.json`](diffhunk://#diff-8abce7b09c25189ac9ac6bee36c93c6f2c05ff2f35cf08f4b96357a5502cea1fR1-R25): This new file provides parameter values for the `existingClusterOnboarding.json` template. It includes placeholders for the AKS resource ID and location, OpenTelemetry logs and metrics ports, and resource tags.

### CODE REVIEW

File: AIOnboardingARMTemplates/existingClusterOnboarding.json [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        "aksResourceId": {
+            "type": "string",
+            "metadata": {
+                "description": "AKS Cluster Resource ID"
+            }
+        },
+        "aksResourceLocation": {
+            "type": "string",
+            "metadata": {
+              "description": "Location of the AKS resource e.g. \"East US\""
```
Comment (tokaplan) [inline AIOnboardingARMTemplates/existingClusterOnboarding.json]: Is this the format we are following? Not "eastus"?

File: AIOnboardingARMTemplates/existingClusterOnboarding.json [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                "contentVersion": "1.0.0.0",
+                "parameters": {},
+                "variables": {},
+                "resources": [
+                    {
+                        "name": "[variables('clusterName')]",
+                        "type": "Microsoft.ContainerService/managedClusters",
+                        "location": "[parameters('aksResourceLocation')]",
+                        "tags": "[parameters('resourceTagValues')]",
+                        "apiVersion": "2024-02-02-preview",
```
Comment (tokaplan) [inline AIOnboardingARMTemplates/existingClusterOnboarding.json:64]: Is this the new _api-version_ that has the latest changes?

File: AIOnboardingARMTemplates/existingClusterOnboarding.json [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+            "defaultValue": "28331"
+        },
+        "openTelemetryMetricsPort": {
+            "type": "int",
+            "metadata": {
+              "description": "Port to use for OpenTelemetry Metrics"
+            },
+            "defaultValue": "28333"
+        },
+        "resourceTagValues": {
```
Comment (tokaplan) [inline AIOnboardingARMTemplates/existingClusterOnboarding.json:31]: What is the usage for these tags?

File: AIOnboardingARMTemplates/existingClusterOnboarding.json (hunk 2 above)
Comment (jatakiajanvi12) [reply AIOnboardingARMTemplates/existingClusterOnboarding.json:64]: yes this is the new version

File: AIOnboardingARMTemplates/existingClusterOnboarding.json (hunk 3 above)
Comment (jatakiajanvi12) [reply AIOnboardingARMTemplates/existingClusterOnboarding.json:31]: I think it is just used by customers to tag their deployments and resources. It is same as Azure tags: https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/tag-resources

File: AIOnboardingARMTemplates/existingClusterOnboarding.json (hunk 1 above)
Comment (jatakiajanvi12) [reply AIOnboardingARMTemplates/existingClusterOnboarding.json]: yes makes sense - my bad. Let me make this update

Comment (tokaplan) [reply AIOnboardingARMTemplates/existingClusterOnboarding.json]: @jatakiajanvi12 let's make sure it's in line with what ARM typically uses. In my experience, templates usually follow the "eastus" format, but my experience is limited.

**Review by tokaplan — APPROVED** [review]

File: AIOnboardingARMTemplates/existingClusterOnboarding.json (hunk 1 above)
Comment (jatakiajanvi12) [reply AIOnboardingARMTemplates/existingClusterOnboarding.json]: yes you are right. Our current ARM templates also follow that and I tested with "eastus" only but there was a description error

**Pull request merged**